deployer/client/client.go

```diff
@@ -1,73 +1,58 @@
 package client
 
 import (
-	"time"
-
 	"github.com/coinbase/step/aws"
 	"github.com/coinbase/step/aws/s3"
 	"github.com/coinbase/step/deployer"
+	"github.com/coinbase/step/machine"
 	"github.com/coinbase/step/utils/to"
 )
 
-// NewRelease returns a valid Release Structure
-func NewRelease(states *string, lambda *string, step *string, bucket *string, region *string, account_id *string, lambda_sha *string) *deployer.Release {
-	r := &deployer.Release{
-		ReleaseId:        to.TimeUUID("release-"),
-		CreatedAt:        to.Timep(time.Now()),
-		LambdaName:       lambda,
-		StepFnName:       step,
-		Bucket:           bucket,
-		StateMachineJSON: to.Strp(to.CompactJSONStr(states)),
-		LambdaSHA256:     lambda_sha,
-	}
-
-	r.SetDefaults(region, account_id)
-	return r
-}
-
-// PrepareRelease returns a release with additional information filled in by querying AWS
-func PrepareRelease(awsc aws.AwsClients, states *string, lambda *string, step *string, bucket *string, zip_file_path *string) (*deployer.Release, error) {
+// PrepareRelease returns a release with additional information filled in
+func PrepareRelease(release *deployer.Release, zip_file_path *string) error {
```

Comment:
i think it may be annoying to pass a struct around functions to be filled up. how do you know when it is fully filled vs partially filled? an idea is to use builder pattern to represent a partially filled deployer.Release, and call .Build() to get the resulting struct. what do you think? i do however like the idea of minimizing the number of parameters on this function, before there were way too many.

Reply:
I dont think it is too necessary at the moment, but if this gets more complicated that pattern might be needed